kubernetes/resource_kubernetes_resource_priority_class.go

```diff
@@ -0,0 +1,227 @@
+				Required:    true,
+				ForceNew:    true,
+			},
+		},
+	}
+}
+
+func resourceKubernetesPriorityClassCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	value := d.Get("value").(int)
+	description := d.Get("description").(string)
+	globalDefault := d.Get("global_default").(bool)
+
+	priorityClass := api.PriorityClass{
+		ObjectMeta:    metadata,
+		Description:   description,
+		GlobalDefault: globalDefault,
+		Value:         int32(value),
+	}
+
+	log.Printf("[INFO] Creating new priority class: %#v", priorityClass)
+	out, err := conn.Scheduling().PriorityClasses().Create(&priorityClass)
+	if err != nil {
+		return fmt.Errorf("Failed to create priority class: %s", err)
+	}
+	log.Printf("[INFO] Submitted new priority class: %#v", out)
+	d.SetId(buildId(out.ObjectMeta))
+
+	err = resource.Retry(1*time.Minute, func() *resource.RetryError {
+		createdPriorityClass, err := conn.Scheduling().PriorityClasses().Get(out.Name, meta_v1.GetOptions{})
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		if createdPriorityClass.Value == priorityClass.Value {
+			return nil
+		}
+		err = fmt.Errorf("Priority class doesn't match after creation.\nExpected: %#v\nGiven: %#v",
+			createdPriorityClass.Value, priorityClass.Value)
+		return resource.RetryableError(err)
+	})
+	if err != nil {
+		return err
+	}
+
+	return resourceKubernetesPriorityClassRead(d, meta)
+}
+
+func resourceKubernetesPriorityClassRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	_, name, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Reading priority class %s", name)
+	priorityClass, err := conn.Scheduling().PriorityClasses().Get(name, meta_v1.GetOptions{})
+	if err != nil {
+		log.Printf("[DEBUG] Received error: %#v", err)
+		return err
+	}
+	log.Printf("[INFO] Received priority class: %#v", priorityClass)
+
+	// This is to work around K8S bug
+	// See https://github.com/kubernetes/kubernetes/issues/44539
+	if priorityClass.ObjectMeta.GenerateName == "" {
+		if v, ok := d.GetOk("metadata.0.generate_name"); ok {
+			priorityClass.ObjectMeta.GenerateName = v.(string)
+		}
+	}
+
+	err = d.Set("metadata", flattenMetadata(priorityClass.ObjectMeta, d))
+	if err != nil {
+		return err
+	}
+
+	err = d.Set("value", priorityClass.Value)
+	if err != nil {
+		return err
+	}
+
+	err = d.Set("description", priorityClass.Description)
+	if err != nil {
+		return err
+	}
+
+	err = d.Set("global_default", priorityClass.GlobalDefault)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceKubernetesPriorityClassUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	_, name, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+
+	ops := patchMetadata("metadata.0.", "/metadata/", d)
+
+	if d.HasChange("value") {
```

Comment:
This should be removed as `value` cannot be updated: https://github.com/terraform-providers/terraform-provider-kubernetes/pull/495#discussion_r303403512

Reply:
Removed.